rest/src/market_data_rest.py

```diff
@@ -0,0 +1,232 @@
+
+    @staticmethod
+    def _nonce():
+        """
+        Nonce counter.
+        :returns: int, an always-increasing unsigned integer (up to 64 bits wide)
+        """
+        return int(1000 * time.time())
+
+    def _query_public(self, method, timeout=10, headers=None, data=None, request_type=GET) -> json:
+        """
+        Used to query the public endpoints
+
+        :param method:
+        :param timeout:
+        :param headers:
+        :param data:
+        :param request_type: str, 'GET or 'POST'
+        :return:
+        """
+        if headers is None:
+            headers = {}
+        if data is None:
+            data = {}
+
+        # the url is composed of the base url + the route to public if any + the endpoint itself
+        url = self.api_url + self.public_path + method
+        # @TODO: remove once we have a logger
+        print(f'\nurl={url}')
+        print(f'\ndata={data}')
+
+        response = make_request(self.session, url, timeout, headers, data, request_type)
+
+        if response.status_code == HTTPStatus.OK:
+            return response.json()
+        else:
+            return self.process_error(response)
+
+    def _query_private(self, method, timeout=10, headers=None, data=None, request_type=GET):
+        """
+        Query private information
+
+        :param method:
+        :param timeout:
+        :param headers:
+        :param data:
+        :param request_type:
+        :return:
+        """
+        if data is None:
+            data = {}
+        data['nonce'] = self._nonce()
+
+        method_path = self.private_path + method
+        url = self.api_url + method_path
+
+        if headers is None:
+            if not self.public_key:
+                raise Exception('Public key is not set!')
+            if not self.private_key:
+                raise Exception('Private key is not set!')
+            signature = self._sign(data, method_path)
+
+            headers = {
+                self.header_key_col: self.public_key,
+                self.header_signature_col: signature
+            }
+
+        response = make_request(self.session, url, timeout, headers, data, request_type)
+        if response.status_code == HTTPStatus.OK:
+            return response.json()
+        else:
+            return self.process_error(response)
+
+    @abstractmethod
+    def _sign(self, data, method_path):
+        """
+        Used to sign request to private endpoints. Each Exchange has its own method of signing
+
+        :param data: dict
+        :param method_path: str
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def format_sym_for_market(self, sym):
+        """
+        Format the sym to the exchange format
+
+        :param sym: str
+        :return: str
+        """
+        pass
+
+    @abstractmethod
+    def format_crypto_back(self, sym):
+        """
+        Format the sym back from the exchange format to our standard forma
+
+        :param crypto:
```

Comment:
sym instead of crypto here, no ?

Reply:
No, crypto actually, so that we know it's not a ccy pair but particularly a crypto ccy